src/components/avatar/Avatar.tsx

```diff
@@ -1,91 +1,103 @@
-                            0 0 0 ${borderSize * 2}px ${theme.palette.primary.main};
-
-                        transform: scale(${scaleFactor}, ${scaleFactor});
-                    }
-                `}
-
-                transition: box-shadow 500ms ease, transform 500ms ease;
-            }
-
-            ${StatusBadge} {
-                position: absolute;
-                bottom: ${AVATAR_SIZE_MAP[size].status.offset}px;
-                right: ${AVATAR_SIZE_MAP[size].status.offset}px;
-            }
-
-            ${MentionBadge} {
-                position: absolute;
-                top: ${mentions === 0 ? -2 : -3}px;
-                right: ${mentions === 0 ? -2 : -3}px;
-            }
-        `;
+import AvatarRoot, { AvatarMentionBadgeRoot, AvatarStatusBadgeRoot } from './Avatar.root';
+
+type PStyledAvatarImage = {
+    image: string;
+};
+
+// TODO@all: not happy with this solution, but it works for now
+//           should be replaced when the Image component is ready
+const StyledAvatarImage = styled.div<PStyledAvatarImage>`
+    flex: 1;
+    align-self: stretch;
+    background-image: url(${(props): string => props.image});
+    background-position: center center;
+    background-repeat: no-repeat;
+    background-size: cover;
+`;
+
+type PLazyAvatarImage = {
+    source: string;
+};
+
+// TODO@all: equally unhappy with this. For now it is ok, but should be changed
+//           once we find a better solution (create a generic LazyImage component?)
+const LazyAvatarImage = ({ source }: PLazyAvatarImage): JSX.Element => {
+    const [image, setImage] = useState('');
+
+    useEffect(() => {
+        Utils.getBase64(source)
+            .then((imageString) => setImage(imageString))
+            .catch(() => {});
+    }, [source]);
+
+    if (!image) {
+        // TODO@all: this is a temporary solution and should be replaced with
+        //           a dedicated `Skeleton` component
+        return <div className={'skeleton'} />;
     }
-);
+
+    return <StyledAvatarImage image={image} />;
+};
+
+const Avatar = ({
+    size = DEFAULT_AVATAR_SIZE,
+    variant = DEFAULT_AVATAR_VARIANT,
+    disableHover = false,
+    isActive = false,
+    onClick,
+    mentions,
+    name,
+    image,
+    status,
+    ...rest
+}: PAvatar): JSX.Element => {
+    const element: 'button' | 'div' = Utils.isFunction(onClick) ? 'button' : 'div';
+
+    // correctness of index is guaranteed by using a tuple for AVATAR_SIZES
+    // - `MentionBadges` are best to be used at size `md` or above (`sm` is the smallest supported size)
+    // - `StatusBadges` are usable on nearly all sizes (except for `xxs` and `xxxs`)
+    const sizeIndex = AVATAR_SIZES.indexOf(size);
+
+    const rootProperties = {
+        size,
+        variant,
+        onClick,
+        disableHover,
+        isActive,
+        ...rest,
+    };
+
+    return (
+        <AvatarRoot
+            as={element}
```

Comment:
I think we should keep the option to pass custom elements to the foundation components but keep it default here in order to avoid handling different edge cases.

Reply:
did the change!